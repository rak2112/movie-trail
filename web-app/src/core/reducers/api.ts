import * as actions from '../actions';
import { Api } from '../interfaces';

export type ApiState = Api | null

export const api = ( state: ApiState = {isFetching: false, hasError: false}, action: actions.ApiActionError): ApiState => {
  
  switch (action.type) {
    case actions.API_REQUEST:
      return { 
        ...state, 
        isFetching: true
      };
    case actions.API_RESPONSE:
      return {
        ...state,
        isFetching: false,
        hasError: false,
        errorMessage: ''
      };
    case actions.API_ERROR:
      return {
        ...state, 
        hasError: true,
        errorMessage: action.error
      };
    default: 
      return state;
  }
};