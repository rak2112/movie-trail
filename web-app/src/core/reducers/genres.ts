import * as actions from '../actions';
import { Genre } from '../interfaces';

export type GenresState = Genre[] | null;

export const genres = ( state: GenresState = [], action: actions.GenreAction): GenresState => {
  if(action.type === actions.LOAD_GENRES_SUCESS) {
    return action.res;
  }
  return state;
};
