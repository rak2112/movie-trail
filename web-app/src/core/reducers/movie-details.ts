import * as actions from '../actions';
import { MovieDetail } from '../interfaces';


export type MovieDetailState = MovieDetail | null;

export const movieDetails = ( state: MovieDetailState =  null, action: actions.MovieDetailAction): MovieDetailState => {
  
  if(action.type === actions.LOAD_MOVIE_DETAILS_SUCCESS) {
    return action.res;
  }

  if(action.type === actions.RESET_MOVIE_DETAILS) {
    return null;
  }
  return state;
};
