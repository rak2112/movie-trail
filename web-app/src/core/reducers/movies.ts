import * as actions from '../actions';
import { Movies } from '../interfaces';

export const intitialMovieState = {
  results: [{
    id: 0,
    backdrop_path: '',
    genre_ids: [],
    genres: [],
    homepage: '',
    overview: '',
    poster_path: '',
    runtime: '',
    release_date: '',
    spoken_languages: [],
    status: '',
    title: '',
    vote_average: ''
  }],
  pageNo: 0,
  totalPages: 0,
  totalResults: 0,
}
export type MoviesState = Movies | null;

export const movies = ( state: MoviesState = intitialMovieState, action: actions.MovieAction): MoviesState => {
  if(action.type === actions.LOAD_MOVIES_SUCESS) {
    return action.res;
  }

  if(action.type === actions.RESET_MOVIES) {
    return intitialMovieState;
  }
  return state;
};
