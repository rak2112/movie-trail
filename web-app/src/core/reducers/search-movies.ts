import * as actions from '../actions';
import { Movie } from '../interfaces';


export type SerachMoviesState = Movie[] | null;

export const searchedMovies = ( state: SerachMoviesState =  null, action: actions.SearchMovieAction): SerachMoviesState => {
  if(action.type === actions.SEARCH_MOVIES_SUCESS) {
    return action.res;
  }
  return state;
};
