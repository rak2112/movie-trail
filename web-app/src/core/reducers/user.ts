import * as actions from '../actions';
import { User } from '../interfaces';

export type UserState = User | null;

export const userReducer = ( state: UserState = null, action: actions.UserAction): UserState => {

  if(action.type === actions.LOG_IN_SUCCESS) {
    return action.res;
  }

  if(action.type === actions.LOG_OUT_SUCESS) {
    return state = null;
  }
  return state;
};

