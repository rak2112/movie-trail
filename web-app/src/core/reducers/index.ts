import { routerReducer } from 'react-router-redux';
import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form'

import { api, ApiState } from './api';
import { genres, GenresState } from './genres';
import { movieDetails, MovieDetailState } from './movie-details';
import { movies, MoviesState } from './movies';
import { user, UserState } from './user';


export interface State {
  readonly api: ApiState;
  readonly genres: GenresState
  readonly movieDetails: MovieDetailState;
  readonly movies: MoviesState;
  readonly user: UserState;
}

const rootReducer = combineReducers({
  api,
  form: formReducer,
  genres,
  movieDetails,
  movies,
  routing: routerReducer,
  user
});

export default rootReducer;