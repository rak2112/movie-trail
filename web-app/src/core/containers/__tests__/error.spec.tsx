import React, { ComponentClass } from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter, Route } from 'react-router';
import { create } from 'react-test-renderer';
import { Store } from 'redux';
import configureMockStore from 'redux-mock-store';

import { State } from '../../reducers';
import ConnectedContainer from '../error';




describe(`Error Container`, () => {
  let store: Store;

  it(`should render`, () => {
    const initialState = { 
      api: {
        hasError: true,
        errorMessage: 'Something has gone wrong!'
      }
    } as State;
    store = configureMockStore()(initialState);

    const component = create(
      <Provider store={store}>
        <ConnectedContainer/>
      </Provider>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

});