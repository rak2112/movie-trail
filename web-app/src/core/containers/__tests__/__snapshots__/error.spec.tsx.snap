// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Container should render 1`] = `
<div
  className="alert alert-danger"
  role="alert"
  style={
    Object {
      "fontSize": "1.8rem",
      "margin": 0,
      "textAlign": "center",
    }
  }
>
  Something has gone wrong!
</div>
`;
